drivers/gpio/gpio_nrfx.c

```diff
@@ -95,6 +106,18 @@ static int gpio_nrfx_pin_configure(const struct device *port, gpio_pin_t pin,
 		return -EINVAL;
 	}
 
+#ifdef CONFIG_SOC_NRF54H20_GPD
+	if (cfg->pad_pd == NRF_GPD_FAST_ACTIVE1) {
+		ret = nrf_gpd_request(NRF_GPD_FAST_ACTIVE1);
+		if (ret < 0) {
+			return ret;
+		}
+
+		retain = nrf_gpio_port_retain_get(cfg->port);
+		nrf_gpio_port_retain_set(cfg->port, retain | BIT(pin));
```

Comment:
This seems to be applied too early. When you apply retention, the following operations like `nrf_gpio_port_out_` or `nrf_gpio_reconfigure()` have no effect on the pins.

The retention should be applied after the pin is fully configured. Right after applying the retention, `nrf_gpd_release()` should be called to allow for low-power states.